src/components/collapsible-motion/collapsible-motion.js

```diff
@@ -117,18 +107,34 @@ class CollapsibleMotion extends React.PureComponent {
       >
         {({ isOpen, toggle }) => (
           <ToggleAnimation isOpen={isOpen} toggle={toggle}>
-            {({
-              containerStyles,
-              toggle: animationToggle,
-              registerContentNode,
-            }) =>
-              this.props.children({
-                isOpen,
-                toggle: animationToggle,
-                containerStyles,
-                registerContentNode,
-              })
-            }
+            {({ animation, toggle: animationToggle, registerContentNode }) => (
+              <ClassNames>
+                {({ css, cx }) => {
+                  let animationClass;
+
+                  if (animation) {
+                    animationClass = css`
+                      animation: ${animation} 200ms forwards;
+                    `;
+                  }
+
+                  const containerClassName = css`
+                    height: ${isOpen ? 'auto' : '0'};
+                    overflow: ${isOpen ? 'hidden' : 'inherit'};
+                  `;
+
+                  return this.props.children({
+                    isOpen,
+                    containerClassName: cx(
```

Comment:
Is the breaking change really necessary? Is there a way to be backward compatible?

Reply:
It would be possible to access the animation name through emotion, but it's not really how they recommend using the library. We could do something like this

```js
 const class = keyframes`
    0% { height: 0; overflow: hidden; }
    99% { height: 90px; overflow: hidden; }
    100% { height: auto; overflow: visible; }
  `;

const animationName = class.name;
const styles = {
   animation: `${class.name} 200ms forwards`
}

```

But if they change the implementation to longer expose this name attribute, I don't know if they would consider this a breaking change or not. I would guess no.

Reply:
This being the `console.log` of the keyframes response

<img width="593" alt="screen shot 2018-12-04 at 12 07 56 pm" src="https://user-images.githubusercontent.com/2425013/49437979-48f9a500-f7bd-11e8-99e1-285d2cba2016.png">

As you can see, the name attribute seems to match the animation name. But would changing the animation name to not match the name attribute be a breaking change? 🤷‍♂️

Reply:
Hm, then we could look into having a `next` branch and merge breaking changes into that? We could continue releases features and fixes from master, and then merge them back into next?

Reply:
I am like being explicit about breaking changes and just making it a quick and short pain point. It feels (to my current understanding) that we can't really work a way around this again. So it'd rather accept it rather than working around it and postponing it. Am I missing something in that?

Reply:
I spoke to @montezume and we can keep the current API (no breaking change) if we accept a bit of weirdness into the component (which is fine for me). We know that there's a way out in case it gets even weirder to keep supporting `containerStyles` but it still feels okay.